client/src/templates/Challenges/classic/mobile-layout.tsx

```diff
@@ -45,6 +45,65 @@ class MobileLayout extends Component<MobileLayoutProps, MobileLayoutState> {
     });
   };
 
+  avoidWindowScrolling = (event: Event) => {
+    event.preventDefault();
+  };
+
+  setToolPanelPosition = () => {
+    const toolPanelGroup = (
+      document.getElementsByClassName(
+        'tool-panel-group-mobile'
+      ) as HTMLCollectionOf<HTMLElement>
+    )[0];
+
+    if (navigator.userAgent.match(/iPhone|Android.+Mobile/)) {
+      if (window.innerHeight > (visualViewport?.height as number)) {
+        setTimeout(() => {
+          window.scrollTo(0, 0);
+
+          toolPanelGroup.style.position = 'absolute';
+          toolPanelGroup.style.bottom =
+            String(window.innerHeight - (visualViewport?.height as number)) +
+            'px';
```

Comment:
You have solved one of the two previous issues, this value still small for my phone.

If it's just my phone, I don't mind pushing this is in.

Reply:
I appreciate for your review and testing again on your mobile phone, @Sboonny.
But I could not identify the cause of the issue which is that `window.innerHeight` and/or `visualViewport?.height` are/is not properly read in resize event on ColorOS/Brave.

The code was tested on my very old iPhone SE(iOS 15.7.1) with Chrome, Firefox, Safari, Opera and Brave but didn't run into the issues which shared in videos above.
Because iPhone's virtual keyboard is opened by tapping an keyboard icon, so If textarea in editor pane is focused, the keyboard is not opened automatically.

Reply:
The issue is in how my device interpret `visualViewport?.height`, as I have mentioned before, it was working before its update 🤷.

This seems to be device specific, which mean it's outside the scope of this PR.